organizations/plugins.go

```diff
@@ -1,269 +1,346 @@
+		return
+	}
+
+	var org Organization
+
+	// convert bson to struct
+	bsonBytes, _ := bson.Marshal(save)
+	err = bson.Unmarshal(bsonBytes, &org)
+
+	if err != nil {
+		utils.GetError(err, http.StatusInternalServerError, w)
+		return
+	}
+
+	var doc map[string]interface{}
+
+	for _, v := range org.Plugins {
+		if v["plugin_id"] == pluginID {
+			doc = v
+			break
+		}
+	}
+
+	if doc == nil {
+		// plugin not found in organization.
+		utils.GetError(err, http.StatusNotFound, w)
+		return
+	}
+
+	utils.GetSuccess("plugin returned successfully", doc, w)
+}
+
+// Remove an organization plugin.
+func (oh *OrganizationHandler) RemoveOrganizationPlugin(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	var orgPlugin OrgPluginBody
+
+	orgID := mux.Vars(r)["id"]
+	pluginID := mux.Vars(r)["plugin_id"]
+
+	err := json.NewDecoder(r.Body).Decode(&orgPlugin)
+	if err != nil {
+		utils.GetError(err, http.StatusBadRequest, w)
+		return
+	}
+
+	// confirm if user_id exists
+	creatorID, err := primitive.ObjectIDFromHex(orgPlugin.UserID)
+
+	if err != nil {
+		utils.GetError(errors.New("invalid user id"), http.StatusBadRequest, w)
+		return
+	}
+
+	user, _ := utils.GetMongoDBDoc(MemberCollectionName, bson.M{"_id": creatorID, "org_id": orgID})
+	if user == nil {
+		utils.GetError(errors.New("member doesn't exist in the organization"), http.StatusBadRequest, w)
+		return
+	}
+
+	var member Member
+	if err = utils.ConvertStructure(user, &member); err != nil {
+		utils.GetError(err, http.StatusInternalServerError, w)
+		return
+	}
+
+	if member.Role != OwnerRole && member.Role != AdminRole {
+		utils.GetError(errors.New("access denied"), http.StatusForbidden, w)
+		return
+	}
+
+	// confirm if user_id exists
+	objID, err := primitive.ObjectIDFromHex(orgID)
+
+	if err != nil {
+		utils.GetError(errors.New("invalid user id"), http.StatusBadRequest, w)
+		return
+	}
+
+	save, _ := utils.GetMongoDBDoc(OrganizationCollectionName, bson.M{"_id": objID})
+
+	if save == nil {
+		utils.GetError(fmt.Errorf("organization %s not found", orgID), http.StatusNotFound, w)
+		return
+	}
+
+	var org Organization
+
+	// convert bson to struct
+	bsonBytes, _ := bson.Marshal(save)
+	err = bson.Unmarshal(bsonBytes, &org)
+
+	if err != nil {
+		utils.GetError(err, http.StatusInternalServerError, w)
+		return
+	}
+
+	pluginIndex := -1
+
+	for k, v := range org.Plugins {
+		if v["plugin_id"] == pluginID {
+			pluginIndex = k
+			break
+		}
+	}
+
+	if pluginIndex == -1 {
+		// plugin not found in organization.
+		utils.GetError(err, http.StatusNotFound, w)
+		return
+	}
+
+	plugins := org.Plugins
+
+	plugins = append(plugins[:pluginIndex], plugins[pluginIndex+1:]...)
+
+	addPlugin := bson.M{"plugins": plugins}
+
+	update, err := utils.UpdateOneMongoDBDoc(OrganizationCollectionName, orgID, addPlugin)
+
+	if err != nil || update.ModifiedCount != 1 {
+		utils.GetError(errors.New("plugin failed to uninstall"), http.StatusBadRequest, w)

```

Comment:
"errors.New("plugin failed to uninstall")" does not really help when there is a problem and we need to debug. You can log the error incase there is one with the logger.